### ISSUE TITLE
Stop putting `lang=` in the url

### ISSUE THREAD

**Issue by thesunshade** [suttacentral/suttacentral#2962]
#### Description:

Currently every page that a user visits has their site language preference dynamically added to their url bar. For example:

```
https://suttacentral.net/iti111/en/sujato?lang=en&layout=plain&reference=none&notes=none&highlight=false&script=latin
```

There is no reason to put *my* language preference into the url that I may share with someone who has a *different* language preference. Doing so breaks the internationalization feature if that parameter is paid attention to.

Currently it seems that for most pages it has no effect anyway, so why not remove it.

The only possible reason to include it is to force viewing a site page (intros and essays) in a specific language. However as I explain [here](https://github.com/suttacentral/suttacentral/issues/2352#issuecomment-1909550147) I think that is also a bad idea. However if we do still want the `lang=` parameter to work for static pages, then we need to stop automatically putting it into the url so someone does not unintentionally share a language specific page.

**Comment by thesunshade** [comment]
I'd especially like to hear from @sabbamitta, @noeismet and @buddhist-uni

**Comment by sabbamitta** [comment]
Under the conditions that
1. localization is working fine and
2. language setting is easy to find

I'd fully support your proposal. Forcing language setting via URL can only be a workaround for a situation where the above two conditions are not fulfilled.

I am going to meet @noeismet in person tomorrow, so we might (among others) have a discussion on this. :slightly_smiling_face:

**Comment by khemarato** [comment]
Yeah, @sabbamitta , I think we should allow users to manually add lang to the URL so that sharing with a tech unsavvy friend takes them to their lang, I just think we shouldn't add the parameter by default to every URL.

**Comment by sabbamitta** [comment]
Yes, I was referring to the *default*.

**Comment by thesunshade** [comment]
> Forcing language setting via URL can only be a workaround for a situation where the above two conditions are not fulfilled.

I mean, I'd even be ok with a "magic" url that would change someone's interface language. But only if that was hidden from everyone but you 🤣.

But really, I see this as a "teach a man to fish" situation. No matter the state of internationalization functioning, it's better to tell them how to change their language.

**Comment by sabbamitta** [comment]
I've in fact tried teaching people how to change their language and found it very hard.

When we were studying DN16 in our discussion group I had to set all their devices to DE each time we met because the site didn't keep the setting (at the time; that may have changed with the introduction of having language setting in the URL by default), and none of them was able to do it, so they asked me to do it.

**Comment by sabbamitta** [comment]
Perhaps I am a bad teacher. But this group is just very tech unsavy. Most of them would never open SC unless they have a link to a specific sutta to click on. They have not much of an idea of the structure of the canon---or no idea actually---and would be totally lost if they had to find their way through the navigation.

There are such people. And they are all wonderful people!

**Comment by thesunshade** [comment]
> I've in fact tried teaching people how to change their language and found it very hard.

1. Click on the three dot menu in the top right corner of the site
2. Click on the first menu item that says "Language"
3. Choose your preferred language.

I'm not trying to me mean, but how could it be any easier than that? Or do you mean change their browser language? I agree that changing your browser language is a bit more technical.

> I had to set all their devices to DE each time we met

Well, that's clearly the bug that needs fixing. But SC  remembers your language setting now doesn't it?

> Most of them would never open SC unless they have a link to a specific sutta to click on.

Yeah, there is a learning curve. Or maybe even cliff.

**Comment by sabbamitta** [comment]
> 
    Click on the three dot menu in the top right corner of the site
    Click on the first menu item that says "Language"
    Choose your preferred language.

Yeah, that's what I thought too. But reality proofed otherwise. :shrug:

**Comment by noeismet** [comment]
> I'd especially like to hear from @sabbamitta, @noeismet and @buddhist-uni

I apologize for not replying sooner, I was driving to meet @sabbamitta. In respect to the localisation/language button, I really think it needs to be more evident. Almost all the people I refer to SuttaCental to come back to me either saying, “but it's only in English”, or “I could not find how to change the language”.  Having the globe just beside the magnifying glass on the home page would certainly make a difference.

**Comment by thesunshade** [comment]
> “I could not find how to change the language”.

Ideally we would be asking them, "So what *did* you try?" I absolutely want to make things as easy as possible for people. I hope that's apparent. However, there is a single, standard menu icon and the very first item in the list says "Language: English". 

> Having the globe just beside the magnifying glass on the home page would certainly make a difference.

I agree. The problem is on mobile. There would be enough room if it only showed on the home page, but after that there is no room. I don't know if a "mobile first" philosophy prevents us from doing it on the desktop version or not.

@noeismet, do you find most people you work with have the browser language set to French?

**Comment by noeismet** [comment]
> I absolutely want to make things as easy as possible for people. I hope that's apparent. However, there is a single, standard menu icon and the very first item in the list says "Language: English".

Oh absolutely, it is apparent. For me, going to the three dots on the top right is very intuitive, and find the menu great. But for people who don't spend as much time in front of the screen as we do, three dots may not mean “menu”.  They may not even spot them as they look for something more straightforward. 

> @noeismet, do you find most people you work with have the browser language set to French?

Yes, definitely. Most people who would want SC in French would have their browser in French.

**Comment by thesunshade** [comment]
> Yes, definitely. Most people who would want SC in French would have their browser in French.

OK, well hopefully Hong Da's work to set SC language based on browser language will help.



> They may not even spot them as they look for something more straightforward.

Maybe you could take a look at this issue https://github.com/suttacentral/suttacentral/issues/2731 regarding the language interface and offer suggestions.

**Comment by noeismet** [comment]
> OK, well hopefully Hong Da's work to set SC language based on browser language will help.

Yes, I think it would, thank you. 

> Maybe you could take a look at this issue #2731 regarding the language interface and offer suggestions.

Ok, will do 🙏
